### PR TITLE
Jetty 12 - Restore ee10 tests

### PR DESCRIPTION
**Pull request by janbartel**
Uncomments and mostly fixes the `jetty-ee10/jetty-ee10-test` subprojects that were commented out. Other issues have been opened for the tests that have been `@Disabled`.

### CODE REVIEW

**Comment by janbartel** [comment]
@lachlan-roberts can you look at this failing test? https://jenkins.webtide.net/blue/organizations/jenkins/jetty.project/detail/PR-9214/7/tests

**Comment by janbartel** [comment]
@lachlan-roberts it would be good if we could get this committed before the next release, can you take a look at the org.eclipse.jetty.http.ByteRangeTest.testBadRangeSetPartiallyBad test.

**Comment by lachlan-roberts** [comment]
@janbartel I removed the test, as we discussed before the test incorrect. It conflicts with other test cases and the RFC says that  

>  "A valid [ranges-specifier](https://www.rfc-editor.org/rfc/rfc9110#rule.ranges-specifier) is "satisfiable" if it contains at least one [range-spec](https://www.rfc-editor.org/rfc/rfc9110#rule.ranges-specifier) that is satisfiable, as defined by the indicated [range-unit](https://www.rfc-editor.org/rfc/rfc9110#range.units). Otherwise, the [ranges-specifier](https://www.rfc-editor.org/rfc/rfc9110#rule.ranges-specifier) is "unsatisfiable"."

so if we have one range-spec which is satisfiable so we should still be able to parse the satisfiable range.

File: jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/handler/EchoHandler.java [hunk 1]
```diff
@@ -29,6 +29,10 @@
  */
 public class EchoHandler extends Handler.Abstract.NonBlocking
 {
+    public EchoHandler()
+    {
+    }
```
Comment (gregw) [inline jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/handler/EchoHandler.java:34]: should be necessary?

File: jetty-ee10/jetty-ee10-tests/jetty-ee10-test-http2-webapp/src/main/java/org/eclipse/jetty/test/webapp/HTTP1Servlet.java [hunk 2]
```diff
@@ -80,7 +80,7 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
         String contextPath = request.getContextPath();
         ServletOutputStream output = response.getOutputStream();
         AsyncContext asyncContext = request.startAsync();
-        http2Client.connect(sslContextFactory, new InetSocketAddress(host, port), new Session.Listener(), new Promise<Session>()
+        http2Client.connect(sslContextFactory, new InetSocketAddress(host, port), new Session.Listener() {}, new Promise<Session>()
```
Comment (gregw) [inline jetty-ee10/jetty-ee10-tests/jetty-ee10-test-http2-webapp/src/main/java/org/eclipse/jetty/test/webapp/HTTP1Servlet.java:83]: strange that a Http1 servlet is using h2?   But it did before so ???

File: jetty-ee10/jetty-ee10-tests/jetty-ee10-test-integration/src/test/resources/deploy.xml [hunk 3]
```diff
@@ -2,30 +2,59 @@
...
+      <Arg>org.eclipse.jetty.ee10.webapp.WebAppContext</Arg>
+    </Call>
+  </Ref>
+
   <Call name="addBean">
     <Arg>
       <New id="DeploymentManager" class="org.eclipse.jetty.deploy.DeploymentManager">
         <Set name="contexts">
-          <Ref refid="WebappContexts" />
+          <Ref refid="contexts" />
```
Comment (gregw) [inline jetty-ee10/jetty-ee10-tests/jetty-ee10-test-integration/src/test/resources/deploy.xml]: It should be capital C Contexts

File: jetty-ee10/jetty-ee10-tests/jetty-ee10-test-integration/src/test/resources/basic-server.xml [hunk 4]
```diff
@@ -19,7 +19,7 @@
   <!-- Set handler Collection Structure                            -->
   <!-- =========================================================== -->
   <Set name="handler">
-    <New id="Contexts" class="org.eclipse.jetty.server.handler.ContextHandlerCollection" />
+    <New id="contexts" class="org.eclipse.jetty.server.handler.ContextHandlerCollection" />
```
Comment (gregw) [inline jetty-ee10/jetty-ee10-tests/jetty-ee10-test-integration/src/test/resources/basic-server.xml]:
```suggestion
    <New id="Contexts" class="org.eclipse.jetty.server.handler.ContextHandlerCollection" />
```

File: jetty-ee10/jetty-ee10-tests/jetty-ee10-test-integration/src/test/resources/add-jetty-test-webapp.xml [hunk 5]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0"?>
 <!DOCTYPE Configure PUBLIC "-//Jetty//Configure//EN" "https://www.eclipse.org/jetty/configure_9_3.dtd">
 
-<Configure id="Contexts" class="org.eclipse.jetty.server.handler.ContextHandlerCollection">
+<Configure id="contexts" class="org.eclipse.jetty.server.handler.ContextHandlerCollection">
```
Comment (gregw) [inline jetty-ee10/jetty-ee10-tests/jetty-ee10-test-integration/src/test/resources/add-jetty-test-webapp.xml]:
```suggestion
<Configure id="Contexts" class="org.eclipse.jetty.server.handler.ContextHandlerCollection">
```

**Review by gregw — CHANGES_REQUESTED** [review]
Only change needed is that the ID is uppercase "Contexts" not "context", so it can match  the `jetty.xml` file.

Other comments are just niggles/questions.

**Review by gregw — APPROVED** [review]

**Pull request merged**